pkg/server/server.go

```diff
@@ -634,6 +634,14 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 	if s.AgentAuthenticationOptions.Enabled {
 		if err := s.authenticateAgentViaToken(stream.Context()); err != nil {
 			klog.ErrorS(err, "Client authentication failed", "agentID", agentID)
+			// Before sending metadata header, returning early will close stream from client side and give any empty value
+			// to client call Header(), which is a bit misleading as the client expect to receive server info to continue
+			// or error to break.
+			h := metadata.Pairs(header.AuthenticationError, err.Error())
```

Comment:
This is the only error type I can think of for now before sending metadata, if we have more types of error in the future, we may think to use a more generalized error type.

Reply:
An alternative approach that would make it easier in the future to reason about backward compatibility: define a new Packet type in konnectivity-client/proto/client/client.proto.

But, it may be nice to keep Packet out of this Connect bootstrap phase and stay with headers only. We could move this definition into proto (enum?), to make the API slightly stronger.

@cheftako WDYT?